hr_expense_product_policy/models/hr_employee.py

```diff
@@ -22,12 +22,11 @@ def _compute_all_allowed_product_ids(self):
         obj_product = self.env["product.product"]
         for employee in self:
             products = employee.allowed_expense_product_ids
-            category_ids = employee.allowed_expense_product_categ_ids.mapped(
-                "id")
+            category_ids = employee.allowed_expense_product_categ_ids.ids
             criteria = [
                 ("categ_id", "in", category_ids),
                 ("hr_expense_ok", "=", True),
-                ]
+            ]
```

Comment:
AFAICT both closing forms are PEP8 compliant.
I personally prefer the indented one, or even better, to have the closing symbol at the end of the block's last line of code.
But this is just a comment, there are no explicit guidelines to enforce here.